docs/control-structures.rst

```diff
@@ -51,40 +56,45 @@ write::
 
 The names of output parameters can be omitted.
 The output values can also be specified using ``return`` statements.
-The ``return`` statements are also capable of returning multiple
-values, see :ref:`multi-return`.
+
+The ``return`` statements are also capable of returning multiple values using ``return (v0, v1, ..., vn)``. The number of parameters must equal the number of output parameters.
+
 Return parameters are initialized to zero; if they are not explicitly
 set, they stay to be zero.
 
 Input parameters and output parameters can be used as expressions in
 the function body.  There, they are also usable in the left-hand side
 of assignment.
 
+...note::
+    You can return all types from Solidity functions, but for structs, only from ``internal`` function calls. You can return ``memory`` structs from any function if you enable the new experimental ``ABIEncoderV2`` feature by adding ``pragma experimental ABIEncoderV2;`` to your source file. This feature introduces other experimental changes that you can read more about in the ABI_ guide.
+
 .. index:: if, else, while, do/while, for, break, continue, return, switch, goto
 
 Control Structures
 ===================
 
-Most of the control structures from JavaScript are available in Solidity
-except for ``switch`` and ``goto``. So
-there is: ``if``, ``else``, ``while``, ``do``, ``for``, ``break``, ``continue``, ``return``, ``? :``, with
-the usual semantics known from C or JavaScript.
+Most of the control structures of curly-bracket languages such as JavaScript or C are available in Solidity except for ``switch`` and ``goto``. The full list is:
 
-Parentheses can *not* be omitted for conditionals, but curly brances can be omitted
-around single-statement bodies.
+- ``if (<condition>) {<true-statement>}``
+- ``if (<condition>) {<true-statement>} else {<false-statement>}``
+- ``while (<condition-true>) {<true-statement>}``
+- ``do {<true-statement>} (while <condition-true>)``
+- ``for (<initial-state>;<final-state>;<iterator>) {<true-statement>}``
+- ``break``
+- ``continue``
+- ``return <value>``
+- ``(if <condition>) ? (<true-statement>) : (<false-statement>)`` (*Conditional or ternary operator, shorthand for ``if else``*)
 
-Note that there is no type conversion from non-boolean to boolean types as
-there is in C and JavaScript, so ``if (1) { ... }`` is *not* valid
-Solidity.
+All with the usual semantics from C or JavaScript, with the following caveats.
 
-.. _multi-return:
+Conditional Values
+------------------
 
-Returning Multiple Values
--------------------------
+You *cannot* omit parentheses for conditionals, but you can omit curly braces around single-statement bodies.
```

Comment:
I'm not too happy with that. The curly braces are not omitted, they are part of the block which is the only way to group statements. Perhaps something like `parentheses for conditionals are *not* optional, but bodies in branches can either be curly-braces-delimited blocks or single staments (without curly braces).`